src/dox/Processor.hx

```diff
@@ -242,6 +242,18 @@ class Processor {
 	function processClassField(path:String, field:ClassField)
 	{
 		field.doc = processDoc(path, field.doc);
+		removeEnumAbstractCast(field);
+	}
+	
+	function removeEnumAbstractCast(field:ClassField)
+	{
+		// remove `cast` from the expression of enum abstract values (#146)
+		if (field.type.match(CAbstract(_, _)) &&
+			field.meta.exists(function(meta) return meta.name == ":impl") &&
+			field.get == RInline && field.set == RNo &&
```

Comment:
I think this might still be too wide. Can you check if the meta also has `@:enum`?